token/program-2022/src/extension/confidential_transfer/processor.rs

```diff
@@ -92,26 +98,127 @@ fn process_update_mint(
     Ok(())
 }
 
+enum ElGamalPubkeySource<'a> {
+    ProofInstructionOffset(i64),
+    ElGamalRegistry(&'a ElGamalRegistry),
+}
+
+/// Processes a [ConfigureAccountWithRegistry] instruction.
+fn process_configure_account_with_registry(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    reallocate: bool,
+) -> ProgramResult {
+    let account_info_iter = &mut accounts.iter();
+    let token_account_info = next_account_info(account_info_iter)?;
+    let _mint_info = next_account_info(account_info_iter)?;
+    let elgamal_registry_account = next_account_info(account_info_iter)?;
+
+    check_elgamal_registry_program_account(elgamal_registry_account.owner)?;
+
+    if reallocate {
+        let payer_info = next_account_info(account_info_iter)?;
+        let system_program_info = next_account_info(account_info_iter)?;
+        reallocate_for_configure_account_with_registry(
+            token_account_info,
+            payer_info,
+            system_program_info,
+        )?;
+    }
+
+    let elgamal_registry_account_data = &elgamal_registry_account.data.borrow();
+    let elgamal_registry_account =
+        pod_from_bytes::<ElGamalRegistry>(elgamal_registry_account_data)?;
+
+    let decryptable_zero_balance = PodAeCiphertext::default();
+    let maximum_pending_balance_credit_counter =
+        DEFAULT_MAXIMUM_PENDING_BALANCE_CREDIT_COUNTER.into();
+
+    process_configure_account(
+        program_id,
+        accounts,
+        &decryptable_zero_balance,
+        &maximum_pending_balance_credit_counter,
+        ElGamalPubkeySource::ElGamalRegistry(elgamal_registry_account),
+    )
+}
+
+fn reallocate_for_configure_account_with_registry<'a>(
+    token_account_info: &AccountInfo<'a>,
+    payer_info: &AccountInfo<'a>,
+    system_program_info: &AccountInfo<'a>,
+) -> ProgramResult {
+    let mut current_extension_types = {
+        let token_account = token_account_info.data.borrow();
+        let account = StateWithExtensions::<Account>::unpack(&token_account)?;
```

Comment:
nit: can you use `PodStateWithExtensions::<PodAccount>` instead for better CU usage?